### PR TITLE
Prepare package for publishing to NPM

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Prepares the project to be published as a public NPM package. 

### What?
To enable wider consumption / adoption of the library and to make it easier to consume for existing users and projects, we would like to publish the package to NPM. This PR prepares the repository fro public consumption, by:
- Cleaning up the README.md
- Adding a LICENSE 
- Cleanup the module exports to include most commonly used interfaces and types in the root
- Add a [Rollup.js](https://rollupjs.org/) script to bundle both commonjs and ESM output modules with compression and adding the LICENSE to the head of the file.

### Outstanding questions:
- What version number should we publish to start? v0.1.0 or v1.0.0?
- Which company should be listed in the LICENSE, currently CBS Interactive Inc.
- Where should we publish the library documentation (probably out of scope for this PR)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Open Insights
+Version: <%= pkg.version %>
+Generated: <%= moment().format("YYYY-MM-DD") %>
+https://github.com/openinsights/openinsights
+
+Copyright (c) <%= moment().format("YYYY") %>, CBS Interactive Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (jakewan) [inline LICENSE:8]: This looks like an MIT license, which sounds good to me. @vnsparacio thoughts?

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Open Insights
+Version: <%= pkg.version %>
+Generated: <%= moment().format("YYYY-MM-DD") %>
```
Comment (jakewan) [inline LICENSE]: I did notice that Moment.js appears to be in legacy mode and they suggest using a different library in most cases ([link](https://github.com/moment/moment#project-status)), though I don't see this as a problem.

File: package.json [hunk 3]
```diff
@@ -1,22 +1,30 @@
 {
-  "name": "open-insights",
+  "name": "@openinsights/openinsights",
```
Comment (jakewan) [inline package.json:2]: @phamann ~This seems like a good idea in terms of namespacing, but I'm not an expert in npm packaging. I looked at some common JS libraries (e.g. lodash, react, chalk, jquery, etc) and didn't come across any that do this. Is it advised?~

👉  Ah, found this documentation on [scope](https://docs.npmjs.com/cli/v7/using-npm/scope). That answers my question. This seems like an especially good idea in case we want to put the generic provider in its own scoped package.

File: package.json [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
-  "repository": "git@github.com:cbsinteractive/open-insights.git",
+  "publishConfig": {
+    "access": "public"
+  },
+  "repository": "git@github.com:openinsights/openinsights.git",
   "license": "MIT",
   "scripts": {
     "build": "tsc --build",
     "build:doc": "rimraf doc && typedoc",
+    "build:release": "rimraf dist && rollup -c",
```
Comment (jakewan) [inline package.json:18]: :point_right: Produces `dist/esm/index.js` using [rollup.js](https://rollupjs.org/guide/en/). When the `-c` flag is used but no config file provided, it uses rollup.config.js by default.

File: LICENSE (hunk 1 above)
Comment (jakewan) [inline LICENSE:3]: This looks like an MIT license, which sounds good to me. @vnsparacio does an MIT license work for the company?

File: package.json [hunk 5]
```diff
@@ -27,13 +35,17 @@
...
+    "rollup-plugin-sizes": "^1.0.3",
+    "rollup-plugin-terser": "^7.0.2",
     "sinon": "^9.0.2",
     "ts-jest": "^26.1.3",
     "typedoc": "^0.17.8",
     "typescript": "^3.9.7"
   },
   "dependencies": {
-    "@fastly/performance-observer-polyfill": "^1.1.1",
+    "@fastly/performance-observer-polyfill": "^2.0.0",
```
Comment (jakewan) [inline package.json:48]: 👉 New stuff!

File: package.json [hunk 6]
```diff
@@ -1,22 +1,30 @@
...
+  "name": "@openinsights/openinsights",
   "version": "0.1.0",
   "main": "dist/index.js",
+  "module": "dist/esm/index.js",
   "types": "dist/types/index.d.ts",
   "files": [
     "dist/**/*"
   ],
-  "repository": "git@github.com:cbsinteractive/open-insights.git",
+  "publishConfig": {
```
Comment (jakewan) [inline package.json:10]: 👉

References:

* https://docs.npmjs.com/cli/v7/configuring-npm/package-json#publishconfig
* https://docs.npmjs.com/cli/v7/using-npm/config#access

**Review by jakewan — APPROVED** [review]
**Conventions**

👉 emoji indicates an informational comment for documentation purposes. No response requested, although feel free to add additional information.

**Outstanding questions:**

>What version number should we publish to start? v0.1.0 or v1.0.0?

I'd vote for v0.1.0 because it feels like we have some work to do in order to make the package more generally useful, e.g. add generic provider. On the other hand, if the generic provider will live in its own scoped package, then perhaps v1.0.0 would be better.

>Which company should be listed in the LICENSE, currently CBS Interactive Inc.

I'm not sure where we are on this, but I hoped we might be able to create some sort of named entity to support this.

>Where should we publish the library documentation (probably out of scope for this PR)

Some ideas:
* https://readthedocs.org
* Github pages
* Github wiki

File: LICENSE (hunk 1 above)
Comment (vnsparacio) [reply LICENSE:3]: I will check in with legal on this

**Comment by vnsparacio** [comment]
Hey Patrick! I've come to some decisions re: the open questions, and would love your feedback on if this seems good to proceed.

**1. What version number should we publish to start? v0.1.0 or v1.0.0?**
At the last meeting we had, we did have a piece of tech debt related to error handling and logging outstanding. I would like for us to revisit if this is something we want to fix _now_, and based on that either get it in ASAP, or go ahead and publish without it. We've had this in the field for months and it's pretty battle tested. We're confident in the core module as is. Therefore, I propose this be version 1.0.0, and we create a separate repository for the generic provider, which will have its own versioning.

**2. Which company should be listed in the LICENSE, currently CBS Interactive Inc.**
We had discussed with legal that we would use ViacomCBS. If you are okay with that, I will confirm with legal that this is still ok, post-merge, and also we should update any instances of CBSi->VCBS

**3. Where should we publish the library documentation (probably out of scope for this PR)**
I would most prefer to have the documentation in markdown on github pages. I'm happy to be open to other suggestions, but if nobody has strong feelings on this, I'm happy to just say let's go with it.

**Review by vnsparacio — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (phamann) [reply package.json:48]: Ah yes, apologies; I should have documented this in the PR description. v2 of the polyfill doesn't introduce any new functionality, however it changes the compilation and output from CommonJS to ESM which makes it easier to consume and bundle in upstream projects such as this - which is why it required a major "breaking change" release. See the notes here: https://github.com/fastly/performance-observer-polyfill/releases/tag/v2.0.0
